File: Daybreak.GWCA/source/payloads/InventoryPayload.cpp
#include <cstdint>
#include <GWCA/GameEntities/Map.h>
#include <json.hpp>
#include <payloads/Bag.h>
#include <payloads/InventoryPayload.h>

using json = nlohmann::json;

namespace Daybreak {
    void to_json(json& j, const InventoryPayload& p) {
        j = json
        {
            {"GoldInStorage", p.GoldInStorage},
            {"GoldOnCharacter", p.GoldOnCharacter},
            {"Backpack", p.Backpack},
            {"BeltPouch", p.BeltPouch},
            {"Bags", p.Bags},
            {"EquipmentPack", p.EquipmentPack},
            {"MaterialStorage", p.MaterialStorage},
            {"UnclaimedItems", p.UnclaimedItems},
            {"StoragePanes", p.StoragePanes},
            {"EquippedItems", p.EquippedItems},
        };
    }
}

File: Daybreak/Services/Scanner/Models/InventoryPayload.cs
using System.Collections.Generic;

namespace Daybreak.Services.Scanner.Models;

internal class InventoryPayload
{
    public uint GoldInStorage { get; set; }
    public uint GoldOnCharacter { get; set; }
    public BagPayload? Backpack { get; set; }
    public BagPayload? BeltPouch { get; set; }
    public BagPayload? EquipmentPack { get; set; }
    public BagPayload? MaterialStorage { get; set; }
    public BagPayload? UnclaimedItems { get; set; }
    public BagPayload? EquippedItems { get; set; }
    public List<BagPayload>? Bags { get; set; }
    public List<BagPayload>? StoragePanes { get; set; }
}
